### PR TITLE
gh-104683: Make Argument Clinic parser template strings class level members

### PR DESCRIPTION
**Pull request by erlend-aasland**

### CODE REVIEW

**Comment by erlend-aasland** [comment]
What do you think of this? IMO, all these static strings pollute `output_templates`. I'm trying to improve the readability of that method.

File: Tools/clinic/clinic.py [hunk 1]
```diff
@@ -873,44 +918,11 @@ def output_templates(
         if new_or_init and not f.docstring:
```
Comment (AlexWaygood) [inline Tools/clinic/clinic.py:918]: There's a `methoddef_define` template string immediately above this line, should that also be moved into the class namespace?

Comment (erlend-aasland) [reply Tools/clinic/clinic.py:918]: Yes, let's do that as well.

Comment (erlend-aasland) [reply Tools/clinic/clinic.py:923]: 62913456b21a64a4f35112d41ade5864a5c7af50

**Review by AlexWaygood — COMMENTED** [review]
Makes sense! You could annotate them with `Final[str]` to indicate to mypy that they're meant to be "constants" that are never changed when accessed on the class or the instance -- they are set once, and then should be treated by the type checker as read-only after that

**Comment by erlend-aasland** [comment]
> Makes sense! You could annotate them with `Final[str]` to indicate to mypy that they're meant to be class-level "constants" that are never changed when accessed on the class or the instance -- they are set once, and then should be treated by the type checker as read-only after that

Neat!

**Pull request closed without merging**